tools/analyze_functions.py

```diff
@@ -0,0 +1,134 @@
+import ast
```

Comment:
add detailed docstring about what it does, and how it is used, give some examples